### DISCUSSION TITLE
HELP ME TO UNDERSTAND BETTER PLEASE .

### DISCUSSION THREAD

**Discussion by kebab03** [Bodmer/TFT_eSPI#1015] [Q&A - Fonts] [answered]
HELLO.
I THOUGHT IN ORDER TO SEND SOME DATA TO THE  SHELD   I NEED TO USE THE FUNCTION "WRITEDATA OR WRITECOMMAND"".BUT WHEN YOU CALL ANY FUNCTION LIKE "

void TFT_eSprite::drawPixel(int32_t x, int32_t y, uint32_t color)
{
  // Range checking
  if ((x < 0) || (y < 0) || !_created) return;

  if (_bpp == 16)
  {
    if ((x >= _iwidth) || (y >= _iheight)) return;
    color = (color >> 8) | (color << 8);
    _img[x+y*_iwidth] = (uint16_t) color;
  }
  else if (_bpp == 8)
  {
    if ((x >= _iwidth) || (y >= _iheight)) return;
    _img8[x+y*_iwidth] = (uint8_t)((color & 0xE000)>>8 | (color & 0x0700)>>6 | (color & 0x0018)>>3);
  }
  else // 1 bpp
  {
    if ((x >= _dwidth) || (y >= _dheight)) return;
    if (_rotation == 1)
    {
      uint16_t tx = x;
      x = _dwidth - y - 1;
      y = tx;
    }
    else if (_rotation == 2)
    {
      x = _dwidth - x - 1;
      y = _dheight - y - 1;
    }
    else if (_rotation == 3)
    {
      uint16_t tx = x;
      x = y;
      y = _dheight - tx - 1;
    }

    if (color) _img8[(x + y * _bitwidth)>>3] |=  (0x80 >> (x & 0x7));
    else       _img8[(x + y * _bitwidth)>>3] &= ~(0x80 >> (x & 0x7));
  }
}

YOU ARE NOT USING   WRITEDATA  !! THEN  HOW   IT IS POSSIBLE TO SEND THE MESSAGE ( "drawPixel") TO  THE LCD ? PLEASE KINDLY EXPLAIN. THANK YOU IN ADVANCE.

**Comment by Bodmer** [comment]
That is a Sprite drawPixel function so it draws the pixel in the RAM buffer. Read [more here](https://github.com/Bodmer/TFT_eSPI#sprites).

**Reply by kebab03** [reply]
Sorry sir , I don't understand how you write data in the display if you don't use any pin to write the data in ram?

**Reply by kebab03** [reply]
I mean how do you transfer data from THE RAM of MCU into the display?

**Comment by Bodmer** [comment, marked as answer]
Use pushSprite(), see examples.

**Comment by Bodmer** [comment]
[This example](https://github.com/Bodmer/TFT_eSPI/tree/master/examples/Sprite/Sprite_draw) is a good place to start.

**Reply by kebab03** [reply]
Thank you for your time.
